pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -253,6 +262,11 @@ func (r *CloudConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			&source.Kind{Type: &configv1.Infrastructure{}},
 			&handler.EnqueueRequestForObject{},
 			builder.WithPredicates(infrastructurePredicates()),
+		).
+		Watches(
+			&source.Kind{Type: &configv1.Network{}},
+			&handler.EnqueueRequestForObject{},
```

Comment:
The `Watches` function requires the `EnqueueRequestForObject` as parameter that is why it was added.

Reply:
From the godoc of that handler:
> EnqueueRequestForObject enqueues a Request containing the Name and Namespace of the object that is the source of the Event

This implies to me it will add a `rencile.Request{Namespace: "", Name: network.Name}`, which is not what we want, we want a reconcile.Request for the objects that this controller works on (configmaps) so we need a map to map the network event to the configmaps that need reconciling

Reply:
We should be using [EnqueueRequestFromMapFunc](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/handler#EnqueueRequestsFromMapFunc) to handle this and the infrastructure object

Reply:
Thanks for the review! I added to both resources and included this in the same commit. Let me know if that is fine by you.